File: 2020 kwiecien/python/2.py
with open("dane4.txt", "r") as f:
    liczby = list(map(int, f.read().strip().split('\n')))

dlugosc, head, tail, i = 0, 0, 0, 0
while i < len(liczby) - 1:
    j, r = i + 1, abs(liczby[i] - liczby[i + 1])
    while j + 1 < len(liczby) and r == abs(liczby[j] - liczby[j + 1]):
        j += 1
    
    if j - i + 1 > dlugosc:
        dlugosc = j - i + 1
        head = liczby[i]
        tail = liczby[j]
    
    i = j
    
print("Długość: ", dlugosc)
print("Wartość na początku:", head)
print("Wartość na końcu:", tail)

File: 2020 kwiecien/c++/2.cpp
#include <iostream>
#include <fstream>
using namespace std;

int a[1005];

int main()
{
    ifstream in("dane4.txt");

    for (int i = 1; i <= 1000; i++)
        in >> a[i];
    
    int i = 1, best = 0, head = 0, tail = 0;
    while (i < 1000)
    {
        int j = i + 1, r = abs(a[i] - a[j]);
        while (j + 1 <= 1000 && r == abs(a[j] - a[j + 1])) j++;

        if (j - i + 1 > best)
        {
            best = j - i + 1;
            head = a[i];
            tail = a[j];
        }

        i = j;
    }
    cout << "Dlugosc: " << best << '\n';
    cout << "Wartosc na poczatku: " << head << '\n';
    cout << "Wartosc na koncu: " << tail << '\n';
}